Goal: do not use MAIL_FILE in userdel:
      When MAIL_FILE is used, the mail spool is in the home directory, so
      it do not need to be removed (in addition to the user's home
      directory)

Note: usermod should be checked also. Maybe MAIL_FILE can be removed from
      login.defs

Index: shadow-4.0.16/src/userdel.c
===================================================================
--- shadow-4.0.16.orig/src/userdel.c	2006-05-31 23:24:50.959789856 +0700
+++ shadow-4.0.16/src/userdel.c	2006-05-31 23:24:57.190842592 +0700
@@ -553,7 +553,7 @@
 
 	maildir = getdef_str ("MAIL_DIR");
 #ifdef MAIL_SPOOL_DIR
-	if (!maildir && !getdef_str ("MAIL_FILE"))
+	if (!maildir)
 		maildir = MAIL_SPOOL_DIR;
 #endif
 	if (!maildir)
